rotkehlchen/api/v1/schemas.py

```diff
@@ -335,61 +312,54 @@ def validate_tx_deletion_schema(
             )
 
 
-class EvmTransactionQuerySchema(
+class TransactionQuerySchema(
         AsyncQueryArgumentSchema,
         TimestampRangeSchema,
 ):
     accounts = fields.List(
-        fields.Nested(RequiredEvmAddressOptionalChainSchema),
+        fields.Nested(RequiredAddressOptionalChainSchema),
         load_default=None,
         validate=webargs.validate.Length(min=1),
     )
-    evm_chain = EvmChainNameField(required=False, load_default=None)
+
+    def __init__(self, database: 'DBHandler') -> None:
+        super().__init__()
+        self.database = database
 
     @validates_schema
-    def validate_evmtx_query_schema(
+    def validate_query_schema(
             self,
             data: dict[str, Any],
             **_kwargs: Any,
     ) -> None:
-        if (
-            data['evm_chain'] is not None and
-            data['evm_chain'] not in get_args(SUPPORTED_CHAIN_IDS)
-        ):
-            raise ValidationError(
-                message=f'rotki does not support evm transactions for {data["evm_chain"]}',
-                field_name='evm_chain',
-            )
-
-    @post_load
-    def make_evm_transaction_query(
-            self,
-            data: dict[str, Any],
-            **_kwargs: Any,
-    ) -> dict[str, Any]:
-        filter_query = EvmTransactionsFilterQuery.make(
-            accounts=data['accounts'],
-            from_ts=data['from_timestamp'],
-            to_ts=data['to_timestamp'],
-            chain_id=data['evm_chain'],
-        )
-
-        return {
-            'async_query': data['async_query'],
-            'filter_query': filter_query,
-        }
-
+        if (accounts := data['accounts']) is None:
+            return
 
-class EvmlikeTransactionQuerySchema(
-        AsyncQueryArgumentSchema,
-        TimestampRangeSchema,
-):
-    accounts = fields.List(
-        fields.Nested(RequiredEvmlikeAddressOptionalChainSchema),
-        load_default=None,
-        validate=webargs.validate.Length(min=1),
-    )
-    chain = StrEnumField(enum_class=EvmlikeChain, load_default=None)
+        with self.database.conn.read_ctx() as cursor:
+            tracked_accounts = self.database.get_blockchain_accounts(cursor)
+
+        tracked_addresses = [
+            address
+            for blockchain in CHAINS_WITH_TRANSACTIONS
+            for address in tracked_accounts.get(blockchain)
+        ]
+        for account in accounts:
+            if account.chain is not None:
+                if account.chain not in CHAINS_WITH_TRANSACTIONS:
```

Comment:
as mentioned above, this `if` check goes away and the nested if becomes simplified